### ISSUE TITLE
WhatsApp web portal doesn't open when our extension is enabled.

### ISSUE THREAD

**Issue by PragneshAgola** [marcosvrs/WTF#27]
There is a problem, WhatsApp web portal doesn't open when our extension is enabled.

There is one more issue continuously below error 
- Uncaught (in promise) Error: Could not establish connection. Receiving end does not exist. & app.ab71bf8ae9b49758a2eb.js:311 - Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'forwardRef') 

Find attached screenshot for reference.

![Screenshot 2023-11-30 at 2 25 52 PM](https://github.com/marcosvrs/WTF/assets/91880452/1ca09776-b383-4aeb-9080-e1025302ccc5)

**Comment by luizsergiorf** [comment]
> There is a problem, WhatsApp web portal doesn't open when our extension is enabled.
> 
> There is one more issue continuously below error
> 
> * Uncaught (in promise) Error: Could not establish connection. Receiving end does not exist.
> * Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'forwardRef')
> 
> Find attached screenshot for reference.
> 
> ![Screenshot 2023-11-30 at 2 25 52 PM](https://private-user-images.githubusercontent.com/91880452/286872928-1ca09776-b383-4aeb-9080-e1025302ccc5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDEzNDM0OTUsIm5iZiI6MTcwMTM0MzE5NSwicGF0aCI6Ii85MTg4MDQ1Mi8yODY4NzI5MjgtMWNhMDk3NzYtYjM4My00YWViLTkwODAtZTEwMjUzMDJjY2M1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzExMzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMTMwVDExMTk1NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY1YjQ0MDMxMTUxZWQ1YjhhZGYyYWEzNTA1MzJjZTYwZjIwYjVmMDFlZDlmMjZiYzA2ODhkNTBlOGExZGIyODUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.bgysnM0oWDC-UEFxCZJQ3odLIZKjfyQnVKKjLPpwtNs)

Hello everything is fine?

I saw this error in the morning, just run the npm update command in the terminal to update the dependencies and recompile the code.

**Comment by luizsergiorf** [comment]
Hello,
I managed to solve the problem on Windows, now on Linux, the problem continues.

**Comment by cstej** [comment]
> Hello, I managed to solve the problem on Windows, now on Linux, the problem continues.

how did you solve it in Windows, I have updated my dependency but still getting the same error. any solution?

**Comment by PragneshAgola** [comment]
@luizsergiorf  & @actions-user can you guide me how to update dependency because I tried it's not working. and I use mac os have any solution?

**Comment by cstej** [comment]
> @luizsergiorf or @cstej can you guide me how to update dependency because I tried it's not working. and I use mac os have any solution?

you can update dependency using `npm update` if using npm or if you are using pnpm then run `pnpm up  --latest`
but I checked this problem this is not specific to window or Linux the issue arises when injecting wa-js, which is required for sending WhatsApp messages."

**Comment by luizsergiorf** [comment]
Hello folks, 

Yesterday by coincidence after giving an npm update, it temporarily resolved. I noticed that it's very random, sometimes the tool opens without errors and sometimes it doesn't. I was thinking, could it be that it's injecting the wa-js without having fully loaded the page. I'm going to do some more tests today, if I find a solution I'll let you know. 

I'm going to post this error in the wa-js project and check if anyone has experienced this.

**Comment by luizsergiorf** [comment]
Folks, it's still not the definitive solution, because it still crashes sometimes when the page loads, but I managed to improve it. change DOMContentLoaded to load in content_script.ts.

```
if (document.readyState === 'complete') {
   injectScript('js/wa-js.js');
} else {
   window.addEventListener('load', () => {
     injectScript('js/wa-js.js');
   });
}
```

**Comment by luizsergiorf** [comment]
> > Folks, it's still not the definitive solution, because it still crashes sometimes when the page loads, but I managed to improve it. change DOMContentLoaded to load in content_script.ts.
> > ```
> > if (document.readyState === 'complete') {
> >    injectScript('js/wa-js.js');
> > } else {
> >    window.addEventListener('load', () => {
> >      injectScript('js/wa-js.js');
> >    });
> > }
> > ```
> 
> I tried your solution but doesn't work means whatsApp web chat does not open properly but whatsApp message sent successfully. Have any idea or it's possible that this actually package issue or whatsApp security changes ?
> 
> ![Screenshot 2023-12-01 at 6 10 36 PM](https://private-user-images.githubusercontent.com/91880452/287266629-2c30163a-72d8-4aef-866e-7416a367e135.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDE0MzY2MDIsIm5iZiI6MTcwMTQzNjMwMiwicGF0aCI6Ii85MTg4MDQ1Mi8yODcyNjY2MjktMmMzMDE2M2EtNzJkOC00YWVmLTg2NmUtNzQxNmEzNjdlMTM1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMDElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjAxVDEzMTE0MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQzZGU3NGUzZWQzZmJkMDFkOGM0MzY1YTU1ZDgyNjQyYzhlMTQ3Zjc4ZTExNWEzNzUyZjQwZDhlMTBiYTFmM2EmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.4lyAmHikGxA7tY4VffCVY643JH3byajuCTRVzihWXBE) ![Screenshot 2023-12-01 at 6 12 37 PM](https://private-user-images.githubusercontent.com/91880452/287266881-87acff48-a029-40bf-a91b-24ddaad302cb.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDE0MzY2MDIsIm5iZiI6MTcwMTQzNjMwMiwicGF0aCI6Ii85MTg4MDQ1Mi8yODcyNjY4ODEtODdhY2ZmNDgtYTAyOS00MGJmLWE5MWItMjRkZGFhZDMwMmNiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMDElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjAxVDEzMTE0MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWUyMjAzODkwNjFmMDljMDlkNjdiMGRkNTI0MzdkNzkxZjgxZTViYTRlYmYxMmY5MWI2YWViMTE1YTMwZjVhZjYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.hrmU7IOSAaYctp3pPbwcnD4Veh00F_uz-sxa7WRYUaE)

Did you reinstall and clear caches?

**Comment by PragneshAgola** [comment]
> > > Folks, it's still not the definitive solution, because it still crashes sometimes when the page loads, but I managed to improve it. change DOMContentLoaded to load in content_script.ts.
> > > ```
> > > if (document.readyState === 'complete') {
> > >    injectScript('js/wa-js.js');
> > > } else {
> > >    window.addEventListener('load', () => {
> > >      injectScript('js/wa-js.js');
> > >    });
> > > }
> > > ```
> > 
> > 
> > I tried your solution but doesn't work means whatsApp web chat does not open properly but whatsApp message sent successfully. Have any idea or it's possible that this actually package issue or whatsApp security changes ?
> > ![Screenshot 2023-12-01 at 6 10 36 PM](https://private-user-images.githubusercontent.com/91880452/287266629-2c30163a-72d8-4aef-866e-7416a367e135.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDE0MzY2MDIsIm5iZiI6MTcwMTQzNjMwMiwicGF0aCI6Ii85MTg4MDQ1Mi8yODcyNjY2MjktMmMzMDE2M2EtNzJkOC00YWVmLTg2NmUtNzQxNmEzNjdlMTM1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMDElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjAxVDEzMTE0MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQzZGU3NGUzZWQzZmJkMDFkOGM0MzY1YTU1ZDgyNjQyYzhlMTQ3Zjc4ZTExNWEzNzUyZjQwZDhlMTBiYTFmM2EmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.4lyAmHikGxA7tY4VffCVY643JH3byajuCTRVzihWXBE) ![Screenshot 2023-12-01 at 6 12 37 PM](https://private-user-images.githubusercontent.com/91880452/287266881-87acff48-a029-40bf-a91b-24ddaad302cb.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDE0MzY2MDIsIm5iZiI6MTcwMTQzNjMwMiwicGF0aCI6Ii85MTg4MDQ1Mi8yODcyNjY4ODEtODdhY2ZmNDgtYTAyOS00MGJmLWE5MWItMjRkZGFhZDMwMmNiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMDElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjAxVDEzMTE0MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWUyMjAzODkwNjFmMDljMDlkNjdiMGRkNTI0MzdkNzkxZjgxZTViYTRlYmYxMmY5MWI2YWViMTE1YTMwZjVhZjYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.hrmU7IOSAaYctp3pPbwcnD4Veh00F_uz-sxa7WRYUaE)
> 
> Did you reinstall and clear caches? Load the page with ctrl+f5 until you can load WhatsApp. Get it working, it's not a permanent solution, but at least I got it running today.

Yes, but face same issue whatsApp portal doesn't open and got this error vendor1~app.faf9096ea5e5e9ae7ce7.js:30 Error: Minified React error #130; visit https://reactjs.org/docs/error-decoder.html?invariant=130&args[]=undefined&args[]= for the full message or use the non-minified dev environment for full errors and additional helpful warnings.

**Comment by PragneshAgola** [comment]
@luizsergiorf & @cstej did you get any solutions?

**Comment by luizsergiorf** [comment]
Hello,

I couldn't find the definitive solution. I am using my temporary solution that I informed you about. If you want to [download ](https://github.com/luizsergiorf/WTF) my project and use it, I made some changes. I haven't done the pull yet because, unfortunately, this solution doesn't resolve it definitively. To make it work, I keep updating the page until it functions without the error. I apologize for not being able to help further.

**Comment by PragneshAgola** [comment]
> Hello,
> 
> I couldn't find the definitive solution. I am using my temporary solution that I informed you about. If you want to [download ](https://github.com/luizsergiorf/WTF) my project and use it, I made some changes. I haven't done the pull yet because, unfortunately, this solution doesn't resolve it definitively. To make it work, I keep updating the page until it functions without the error. I apologize for not being able to help further.

Yes I am use your project , Can you guide me which change have you done?

**Comment by PragneshAgola** [comment]
Thanks @luizsergiorf , I have also changed according to you and now we can send bulk messages with wa number identity validation. But still there is a problem, whatsApp web page appears blank. have any idea how solve it?

![Screenshot 2023-12-06 at 11 40 33 AM](https://github.com/marcosvrs/WTF/assets/91880452/0234faf4-02d1-4590-8f90-0ccfe5c66ef0)

**Comment by marcosvrs** [comment]
@luizsergiorf @PragneshAgola Thank you very much for your support and open discussion about it. I appreciate how our community is growing!

I have found a workaround to it. I don't know how long it will work, since it's based on a class added to the html element, and it might change at any time. But at least we can work with it until either we find a better solution or wppconnect provides us another approach to inject.

That said, please take some time to review and test the code here:
https://github.com/marcosvrs/WTF/pull/28

**Comment by luizsergiorf** [comment]
Hello folks,

I appreciate @marcosvrs and @PragneshAgola very much. Congratulations to everyone for the work!

I was wondering if it would be valid to inject this when manually opening the extension in Google Chrome. This way, we would only need to wait for WhatsApp Web to load completely. What do you think?

**Comment by PatelBassar** [comment]
@PragneshAgola  @marcosvrs @luizsergiorf  anyone of you can updated zip file to upload on unpacked extension part. I dont know how to compile this code. TIA

**Comment by marcosvrs** [comment]
@PatelBassar it got deployed 😉

**Comment by marcosvrs** [comment]
I will be tracking the issue https://github.com/wppconnect-team/wa-js/issues/1521 to check when there is a final solution provided by the library. So far, we will continue with the workaround.

Please let me know if there is any issue with the workaround provided.

**Comment by Himanshusha673** [comment]
@marcosvrs did you find the solution yet?

**Comment by Himanshusha673** [comment]
@marcosvrs @luizsergiorf @PragneshAgola  i am unable to send messages can anyone please help me i am new in chrome extension development, when web page loads i am still encounter this issue "web.whatsapp.com/:1 Uncaught (in promise) Error: Could not establish connection. Receiving end does not exist."
